lib/SILGen/SILGenPoly.cpp

```diff
@@ -7012,8 +7033,61 @@ SILGenFunction::emitVTableThunk(SILDeclRef base,
     }
   }
 
+  // Now that we have translated arguments and inserted our thunk indirect
+  // parameters... before we forward those arguments, insert the implicit
+  // leading parameter.
+  if (options.contains(ThunkGenFlag::CalleeHasImplicitIsolatedParam)) {
+    auto baseIsolation =
+        swift::getActorIsolation(base.getAbstractFunctionDecl());
+    switch (baseIsolation) {
+    case ActorIsolation::Unspecified:
+    case ActorIsolation::Nonisolated:
+    case ActorIsolation::NonisolatedUnsafe:
+      args.push_back(emitNonIsolatedIsolation(loc).getValue());
+      break;
+    case ActorIsolation::Erased:
+      llvm::report_fatal_error("Found erased actor isolation?!");
```

Comment:
Can we fail more gracefully here, by reporting a normal error and sticking in an under or something?